### ISSUE TITLE
Allow for passing a random state to test objects and use rng instead of np.random

### ISSUE THREAD

**Issue by bdpedigo** [neurodata/hyppo#192]
**Is your feature request related to a problem? Please describe.**
I don't see any way to control the random state for most tests (e.g. [dcorr](https://hyppo.neurodata.io/api/generated/hyppo.independence.dcorr#hyppo.independence.Dcorr))

Is there some way that I am missing? 

**Describe the solution you'd like**
Be able to pass a random state/seed to the independence (or other) test object to enable reproducibility

**Describe alternatives you've considered**
setting the global numpy random state - but I'm not sure how this plays with parallelization, different platforms, etc. and have found this to be unreliable in the past.

**Example**
https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html as an example of passing a random state

**Comment by sampan501** [comment]
This is a good issue, random state currently can't be set in any of the methods in hyppo, but probably should. I'm editing this issue to upgrade to numpy random to 1.20. Will probably need to update joblib using something like this: https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html

**Comment by bdpedigo** [comment]
> This is a good issue, random state currently can't be set in any of the methods in hyppo, but probably should. 

Sweet!

> I'm editing this issue to upgrade to numpy random to 1.20. 

Do you mean you only want to allow 1.20? If so, why? I know they have some new randomness stuff and only want you to use `rng.` methods going forward but does that preclude using older versions?

> Will probably need to update joblib using something like this: https://joblib.readthedocs.io/en/latest/auto_examples/parallel_random_state.html

Yup! That's what we're doing. 

@kareef928 this is ultimately the bottleneck for https://github.com/microsoft/graspologic/issues/775 in case you are interested or just curious.

**Comment by sampan501** [comment]
> Do you mean you only want to allow 1.20? If so, why? I know they have some new randomness stuff and only want you to use `rng.` methods going forward but does that preclude using older versions?

Not really, just giving myself a reminder that I want to eventually upgrade everything to numpy 1.20+ and drop python 3.6 support. I'll probably add a minor version that supports using `np.random` and a major version that supports `rng.`

**Comment by bdpedigo** [comment]
> Not really, just giving myself a reminder that I want to eventually upgrade everything to numpy 1.20+

I guess I'm just saying I don't see why one would need to do this as the `rng.` stuff is supported in the older versions of numpy

**Comment by sampan501** [comment]
> I guess I'm just saying I don't see why one would need to do this as the `rng.` stuff is supported in the older versions of numpy

Ah, I got the numpy version wrong. I think that `rng.` started getting supported numpy 1.17+. Changed the issue to reflect that I'm changing `np.random` to a generator instance.

**Comment by TacticalFallacy** [comment]
I am interested in contributing to this!

**Comment by sampan501** [comment]
@TacticalFallacy this isn't really enough for an NDD project, since it is really a simple fix. I would look more at issues with the `ndd` label first

**Comment by bdpedigo** [comment]
@sampan501 how would you feel about @kareef928 working on this? I'd like to be able to fix https://github.com/microsoft/graspologic/issues/775

**Comment by sampan501** [comment]
sounds good!

**Comment by kareef928** [comment]
cool!
